packages/captp/src/captp.js

```diff
@@ -249,59 +318,66 @@ export const makeCapTP = (
         if (unplug !== false) {
           return quietReject(unplug);
         }
-        const [questionID, pr] = makeQuestion();
+        const [questionID, promise] = makeQuestion();
         send({
           type: 'CTP_CALL',
           epoch,
           questionID,
           target,
           method: serialize(harden([prop])),
         });
-        return harden(pr.p);
+        return promise;
       },
       applyFunction(_o, args) {
         if (unplug !== false) {
           return quietReject(unplug);
         }
-        const [questionID, pr] = makeQuestion();
+        const [questionID, promise] = makeQuestion();
         send({
           type: 'CTP_CALL',
           epoch,
           questionID,
           target,
           method: serialize(harden([null, args])),
         });
-        return harden(pr.p);
+        return promise;
       },
       applyMethod(_o, prop, args) {
         if (unplug !== false) {
           return quietReject(unplug);
         }
         // Support: o~.[prop](...args) remote method invocation
-        const [questionID, pr] = makeQuestion();
+        const [questionID, promise] = makeQuestion();
         send({
           type: 'CTP_CALL',
           epoch,
           questionID,
           target,
           method: serialize(harden([prop, args])),
         });
-        return harden(pr.p);
+        return promise;
       },
     };
 
-    const pr = {};
-    pr.p = new HandledPromise((res, rej, resolveWithPresence) => {
-      pr.rej = rej;
-      pr.resPres = () => resolveWithPresence(handler);
-      pr.res = res;
-    }, handler);
+    /** @type {Settler | undefined} */
+    let settler;
+    const promise = new HandledPromise(
+      (resolve, reject, resolveWithPresence) => {
+        settler = Far('settler', {
+          resolve,
+          reject,
+          resolvePresence: () => resolveWithPresence(handler),
+        });
+      },
+      handler,
+    );
+    assert(settler);
 
     // Silence the unhandled rejection warning, but don't affect
     // the user's handlers.
-    pr.p.catch(e => quietReject(e, false));
+    promise.catch(e => quietReject(e, false));
 
-    return harden(pr);
+    return harden({ promise, settler });
```

Comment:
If you did declare a `RemoteKit` type to be this record's type, I'd probably find myself using this elsewhere. Just sayin.